financial-connections/src/main/java/com/stripe/android/financialconnections/features/consent/ConsentViewModel.kt

```diff
@@ -8,26 +8,44 @@ import com.stripe.android.financialconnections.di.financialConnectionsSubCompone
 import com.stripe.android.financialconnections.domain.AcceptConsent
 import com.stripe.android.financialconnections.domain.NativeAuthFlowCoordinator
 import com.stripe.android.financialconnections.domain.NativeAuthFlowCoordinator.Message.RequestNextStep
-import com.stripe.android.financialconnections.domain.NativeAuthFlowCoordinator.Message.UpdateManifest
 import com.stripe.android.financialconnections.features.consent.ConsentState.ViewEffect.OpenUrl
-import com.stripe.android.financialconnections.model.FinancialConnectionsSessionManifest
 import com.stripe.android.financialconnections.navigation.NavigationDirections
 import com.stripe.android.financialconnections.presentation.FinancialConnectionsUrls
+import com.stripe.android.financialconnections.repository.FinancialConnectionsManifestRepository
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 internal class ConsentViewModel @Inject constructor(
     initialState: ConsentState,
     private val acceptConsent: AcceptConsent,
     private val nativeAuthFlowCoordinator: NativeAuthFlowCoordinator,
+    private val repository: FinancialConnectionsManifestRepository,
     private val logger: Logger
 ) : MavericksViewModel<ConsentState>(initialState) {
 
+    init {
+        viewModelScope.launch {
+            val manifest = repository.getOrFetchManifest()
+            setState {
+                copy(
+                    disconnectUrl = ConsentUrlBuilder.getDisconnectUrl(manifest),
+                    faqUrl = ConsentUrlBuilder.getFAQUrl(manifest),
+                    dataPolicyUrl = ConsentUrlBuilder.getDataPolicyUrl(manifest),
+                    stripeToSUrl = ConsentUrlBuilder.getStripeTOSUrl(manifest),
+                    privacyCenterUrl = ConsentUrlBuilder.getPrivacyCenterUrl(manifest),
+                    title = ConsentTextBuilder.getConsentTitle(manifest),
+                    bullets = ConsentTextBuilder.getBullets(manifest),
+                    requestedDataTitle = ConsentTextBuilder.getDataRequestedTitle(manifest),
+                    requestedDataBullets = ConsentTextBuilder.getRequestedDataBullets(manifest)
+                )
+            }
+        }
+    }
+
     fun onContinueClick() {
         viewModelScope.launch {
-            val manifest: FinancialConnectionsSessionManifest = acceptConsent()
+            acceptConsent()
             with(nativeAuthFlowCoordinator()) {
-                emit(UpdateManifest(manifest))
```

Comment:
We don't need to communicate the activity the new manifest to update the cached version anymore. Instead the repository caches the new one after accepting consent.